flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterGateway.java

```diff
@@ -218,4 +218,12 @@ void failSlot(final ResourceID taskManagerId,
 			final TaskManagerLocation taskManagerLocation,
 			final UUID leaderId,
 			@RpcTimeout final Time timeout);
+
+	/**
+	 * Send the heartbeat to job manager from task manager
+	 *
+	 * @param resourceID unique id of the task manager
+	 * @param payload the payload information from the task manager
+	 */
+	void heartbeatFromTaskManager(final ResourceID resourceID, final Object payload);
```

Comment:
Currently we have not considered which specific payload information should be attached with the heartbeat, so use the object to work around.  It should be changed to specific type if confirmation. Do you think which payload should be attached necessary currently?